### ISSUE TITLE
Claim BTG with private key of BTC, need new BTG wallet

### ISSUE THREAD

**Issue by jphellemons** [BTCGPU/BTCGPU#294]
#### If so, please list the steps to reproduce below:
1. Bitcoin Gold version v0.15.0.2 (64-bit) (full node) for windows
2. complete sync
3. made a wallet at installation

### Expected behaviour
I wanted to restore and claim my btg with my btc private key

### Actual behaviour
can't find an option for it. renamed wallet.dat to wallet-old.dat and see if it would force to create a new one or restore one. but didn't

Please let me know if I need to provide more info.

**Comment by pocesar** [comment]
you need to dump your priv keys for the addresses that had funds before the fork, then importprivkey on BTG

**Comment by jphellemons** [comment]
Could you explain a bit more about "importprivkey on BTG"

I have the old address, old seed words, old wallet password, old private keys etc. I just can't find the option in the BTG client to make a new wallet or import stuff etc. It just starts syncing the network and once the sync is done, it starts without a password and opens a wallet which I can't close or switch wallets.

**Comment by jphellemons** [comment]
I have found the cli in "C:\Program Files\BitcoinGold\daemon\bgold-cli.exe" and found this topic: https://github.com/BTCGPU/BTCGPU/issues/131 
When I run the cli with the importprivkey and no other options (to see what is available) I get this:

````
C:\Program Files\BitcoinGold\daemon>bgold-cli.exe importprivkey
error: Could not locate RPC credentials. No authentication cookie could be found, and no rpcpassword is set in the configuration file (C:\Users\myuser\AppData\Roaming\BitcoinGold\bitcoingold.conf)
````

**Comment by pocesar** [comment]
you can do this inside your client, using Help > Debug Window, and going to the Console tab. there, you'll need to provide your password for your wallet (if there's any), so you can issue dumpprivkey

**Comment by jphellemons** [comment]
Somehow my "AppData\Roaming\BitcoinGold" is empty. Is this a problem?
I have the full blockchain on an external drive. And I think that I have my priv key of btc before the fork. I have the seed and address of the pre-fork. I had an electrum wallet.

**Comment by jphellemons** [comment]
I did "importprivkey myprivatekeyfrombtc" in the debug console in the client and it is now rescanning. 1% it's very very slow. Hope it works!

**Comment by pocesar** [comment]
oh, forgot about that... you need to execute it as `importprivkey "addr" "label" false`, as the `false` at the end won't rescan the whole blockchain (it will take HOURS)

**Comment by jphellemons** [comment]
and addr is the private key? and what for is the label?

**Comment by jphellemons** [comment]
`importprivkey "myprivatekey" "label" false` gives `null`

**Comment by pocesar** [comment]
yes, that's the expected "success" return. once you're done importing everything, close the client and reopen it with `-rescan` in command line.

**Comment by jphellemons** [comment]
So I do need to rescan and wait for x hours? How do I know that I have imported the right private key? I took the private key of the old btc address. But in btgexplorer.com it shows that my btg is in another castle and I do not have the private key (yet) of that address. All my addresses (when I export the keys from electrum) start with a 1 and according to btgexplorer it starts with a G. ppf it is hard with all those addresses and private keys. I did manage to claim my BitcoinCash which also took me several tries because I moved from multibit to electrum a while ago.

**Comment by pocesar** [comment]
yes, it's a manual task, unless you automate it with a script. I faced this same problem and needed something for a dump from pywallet, you might want to change it and fit your needs https://github.com/pocesar/node-importprivkey-pywallet
but new versions of the cli comes with dumpwallet, and generate a non-standard wallet dump for importing, which might be easier

**Comment by jphellemons** [comment]
Update: The rescan is completed. I have restored something. I have 3 transactions. One incoming and two outgoing, leaving me with an empty wallet. This is perhaps the wallet I had with the BTC which I cleared as was recommended claiming the Bitcoin Cash. So I have to export private keys of my current BTC wallet and import that key and start over I think. Is there a risk that I will lose BTC. They told me during the BCC claim...
